### PR TITLE
🌱 Consumes upstream image-registry-operator-api module

### PR DESCRIPTION
**Pull request by srm09**
**What does this PR do, and why is it needed?**
Replaces the hardcoded image-registry API types with the upstreamed API types for the image-registry-operator

**Which issue(s) is/are addressed by this PR?**:
n/a

**Are there any special notes for your reviewer**:
The generated CRDs under `config/crds/external` have been updated to be in sync with the latest from the upstream API types.

**Please add a release note if necessary**:
```release-note
🌱 Consumes upstream image-registry-operator-api module
```

### CODE REVIEW

**Comment by zyiyi11** [comment]
LGTM. I triggered https://vmoperator.svc.eng.vmware.com/view/VM%20Operator/job/cayman-vmoperator-pr/109/ for a fast e2e run.

**Review by zyiyi11 — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -39,6 +39,8 @@ require (
 	sigs.k8s.io/yaml v1.3.0
 )
 
+require github.com/vmware-tanzu/image-registry-operator-api v0.0.0-20230526154708-f67dac7c805f
```
Comment (aruneshpa) [inline go.mod:42]: Any reason for not moving it to the current require block around L21 (govmomi, ncp, vmop/api are all there, so I think this belongs there)?

File: test/builder/util.go [hunk 2]
```diff
@@ -18,12 +18,12 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 
-	imgregv1a1 "github.com/vmware-tanzu/vm-operator/external/image-registry/api/v1alpha1"
-	topologyv1 "github.com/vmware-tanzu/vm-operator/external/tanzu-topology/api/v1alpha1"
+	imgregv1a1 "github.com/vmware-tanzu/image-registry-operator-api/api/v1alpha1"
```
Comment (aruneshpa) [inline test/builder/util.go:21]: Any reason for splitting these two (ncp and img reg)?

**Review by aruneshpa — APPROVED** [review]
Two very minor nits.  LGTM.

File: go.mod (hunk 1 above)
Comment (srm09) [reply go.mod:42]: Nothing in particular, `go mod` must have added it in that place. Refactored.

**Pull request merged**